src/components/Search/index.tsx

```diff
@@ -281,7 +281,8 @@ function Search({queryJSON}: SearchProps) {
     };
 
     const openReport = (item: TransactionListItemType | ReportListItemType | ReportActionListItemType) => {
-        let reportID = SearchUtils.isTransactionListItemType(item) && !item.isFromOneTransactionReport ? item.transactionThreadReportID : item.reportID;
+        const isFromSelfDM = item.reportID === CONST.REPORT.UNREPORTED_REPORTID;
+        let reportID = SearchUtils.isTransactionListItemType(item) && (!item.isFromOneTransactionReport || isFromSelfDM) ? item.transactionThreadReportID : item.reportID;
```

Comment:
For some old transactions, the `transactionThreadReportID` is 0, we also need to add condition to avoid that case here.
More details in this comment: https://github.com/Expensify/App/issues/59348#issuecomment-2788802840